esp-hal/src/mcpwm/mod.rs

```diff
@@ -181,6 +181,7 @@ impl<'d, PWM: PwmPeripheral> McPwm<'d, PWM> {
             operator0: Operator::new(),
             operator1: Operator::new(),
             operator2: Operator::new(),
+            _guard: guard,
```

Comment:
I think each Operator (and Timer, I guess) needs its own guard, otherwise it's too easy to accidentally drop the peripheral.

Reply:
Nevermind, I didn't know Operators have guards, in this case this outer one isn't necessary.

Reply:
they all include guards